src/connection.ts

```diff
@@ -886,39 +886,33 @@ export class Connection extends EventEmitter {
 
     if (responsePacket) {
       this.handleFrames(responsePacket.frames)
-      if (responsePacket.prepareAmount.isGreaterThan(0)
-        && amountToSend.isGreaterThan(0)) {
-        this._lastPacketExchangeRate = responsePacket.prepareAmount.dividedBy(amountToSend)
-      }
-    }
 
-    if (!responsePacket || responsePacket.ilpPacketType === IlpPacketType.Reject) {
-      // Handle reject
-      this.debug(`packet ${requestPacket.sequence} was rejected`)
+      if (responsePacket.ilpPacketType === IlpPacketType.Fulfill) {
+        for (let stream of streamsSentFrom) {
+          stream._executeHold(requestPacket.sequence.toString())
+        }
 
-      // Put money back into MoneyStreams
-      for (let stream of streamsSentFrom) {
-        stream._cancelHold(requestPacket.sequence.toString())
-      }
-    } else {
-      for (let stream of streamsSentFrom) {
-        stream._executeHold(requestPacket.sequence.toString())
-      }
-      this._totalDelivered = this._totalDelivered.plus(responsePacket.prepareAmount)
-      this._totalSent = this._totalSent.plus(amountToSend)
-
-      // If we're trying to pinpoint the Maximum Packet Amount, raise
-      // the limit because we know that the testMaximumPacketAmount works
-      if (this.maximumPacketAmount.isFinite()
-        && amountToSend.isEqualTo(this.testMaximumPacketAmount)
-        && this.testMaximumPacketAmount.isLessThan(this.maximumPacketAmount)) {
-        const newTestMax = this.maximumPacketAmount.plus(this.testMaximumPacketAmount).dividedToIntegerBy(2)
-        this.debug(`maximum packet amount is between ${this.testMaximumPacketAmount} and ${this.maximumPacketAmount}, trying: ${newTestMax}`)
-        this.testMaximumPacketAmount = newTestMax
-      }
+        // Update stats based on amount sent
+        if (responsePacket.prepareAmount.isGreaterThan(0)
+          && amountToSend.isGreaterThan(0)) {
+          this._lastPacketExchangeRate = responsePacket.prepareAmount.dividedBy(amountToSend)
```

Comment:
If we only calculate this on the Fulfill then the lastPacketExchangeRate is only calculated on Fulfill. Previously a Reject would still calculate it. @sharafian thoughts?